### PR TITLE
Restructure 1/2

### PR DESCRIPTION
**Pull request by LinhWuerzburger**
changed internal structure of multigrid.

GPU lists are now stored in classes which makes it easier to store the necessary parameters like the size of the array and starting index of a specific object/level.

### CODE REVIEW

**Review by ChristianVonMach — CHANGES_REQUESTED** [review]

File: gtests/Environment.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/// \file       Environment.cpp
```
[the remark this answers is not in the document]
Comment (LinhWuerzburger) [reply gtests/Environment.cpp:1]: it is not part of src files, as it is only a class to be executed first for gtest

File: gtests/boundary/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+target_sources(artss_gtest
+        PRIVATE
```
[the remark this answers is not in the document]
Comment (LinhWuerzburger) [reply gtests/boundary/CMakeLists.txt]: 05ea776

File: gtests/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,11 @@
 add_executable(artss_gtest ${ARTSS_SOURCE_FILES})
 
+target_sources(artss_gtest
+        PRIVATE
```
[the remark this answers is not in the document]
Comment (LinhWuerzburger) [reply gtests/CMakeLists.txt]: 05ea776

File: src/analysis/Analysis.cpp [hunk 4]
```diff
@@ -119,23 +117,22 @@ bool Analysis::compare_solutions(read_ptr num, read_ptr ana, FieldType type, rea
...
-    real r;
 
     auto boundary = BoundaryController::getInstance();
-    size_t *inner_list = boundary->get_inner_list_level_joined();
-    size_t size_inner_list = boundary->get_size_inner_list();
+    size_t *inner_list = boundary->get_domain_inner_list_level_joined();
+    size_t size_inner_list = boundary->get_size_domain_inner_list_level_joined(0);
 
-    // weighted 2-norm
+    //// weighted 2-norm
```
[the remark this answers is not in the document]
Comment (LinhWuerzburger) [reply src/analysis/Analysis.cpp]: cdadf16

File: src/utility/Utility.cpp [hunk 5]
```diff
@@ -8,17 +8,33 @@
...
+std::vector<size_t> get_coordinates(size_t index, size_t Nx, size_t Ny) {
+    std::vector<size_t> coordinates;
+    coordinates.reserve(3);
+    size_t k = getCoordinateK(index, Nx, Ny);
+    size_t j = getCoordinateJ(index, Nx, Ny, k);
+    size_t i = getCoordinateI(index, Nx, Ny, j, k);
+    coordinates.push_back(i);
+    coordinates.push_back(j);
+    coordinates.push_back(k);
+    return coordinates;
```
[the remark this answers is not in the document]
Comment (LinhWuerzburger) [reply src/utility/Utility.cpp]: 36da2eb

File: src/utility/Utility.cpp [hunk 6]
```diff
@@ -8,17 +8,33 @@
...
 #include <clocale>
 #endif
 
 
 namespace Utility {
+    static std::string class_name = "Utility";
+    static std::string global_logger = "ARTSS";
+
+    // do not use only for debug purpose
+std::vector<size_t> get_coordinates(size_t index, size_t Nx, size_t Ny) {
```
[the remark this answers is not in the document]
Comment (LinhWuerzburger) [reply src/utility/Utility.cpp]: 36da2eb

**Review by ChristianVonMach — APPROVED** [review]

**Pull request merged**